modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/display/context/ShippingMethodCheckoutStepDisplayContext.java

```diff
@@ -66,6 +73,36 @@ public CommerceOrder getCommerceOrder() {
 		return _commerceOrder;
 	}
 
+	public List<CommerceShippingFixedOption> getCommerceOrderFiltered()
+		throws PortalException {
+
+		List<CommerceShippingFixedOption> commerceShippingFixedOptions =
+			new ArrayList<>();
+		List<CommerceShippingMethod> commerceShippingMethods =
+			getCommerceShippingMethods();
+		CommerceOrder commerceOrder = getCommerceOrder();
+
+		for (CommerceShippingMethod commerceShippingMethod :
+				commerceShippingMethods) {
+
+			List<CommerceShippingFixedOption> commerceShippingFixedOptionList =
+				_commerceShippingFixedOptionLocalService.
+					getCommerceOrderTypeCommerceShippingFixedOptions(
+						commerceOrder.getCompanyId(),
+						commerceOrder.getCommerceOrderTypeId(),
+						commerceShippingMethod.getCommerceShippingMethodId());
+
+			if ((commerceShippingFixedOptionList != null) &&
+				!ListUtil.isEmpty(commerceShippingFixedOptionList)) {
```

Comment:
my bad I did not explain that, but the ListUtil.isEmpty already check for null and empty, so you don't need to do it 2 times